Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/DateExtractorChs.cs

```diff
@@ -1,220 +1,218 @@
-
-        private static readonly IParser _integerParser =
-            AgnosticNumberParserFactory.GetParser(AgnosticNumberParserType.Integer,
-                new ChineseNumberParserConfiguration());
-
-        private static readonly DurationExtractorChs _durationExtractor = new DurationExtractorChs();
-
-        public List<ExtractResult> Extract(string text)
-        {
-            var tokens = new List<Token>();
-            tokens.AddRange(BasicRegexMatch(text));
-            tokens.AddRange(ImplicitDate(text));
-            tokens.AddRange(DurationWithBeforeAndAfter(text));
-
-            return Token.MergeAllTokens(tokens, text, ExtractorName);
-        }
-
-        public List<Token> ExtractRaw(string text)
-        {
-            var tokens = new List<Token>();
-            tokens.AddRange(BasicRegexMatch(text));
-            tokens.AddRange(ImplicitDate(text));
-
-            return tokens;
-        }
-
-        // match basic patterns in DateRegexList
-        private List<Token> BasicRegexMatch(string text)
-        {
-            var ret = new List<Token>();
-            foreach (var regex in DateRegexList)
-            {
-                var matches = regex.Matches(text);
-                foreach (Match match in matches)
-                {
-                    ret.Add(new Token(match.Index, match.Index + match.Length));
-                }
-            }
-            return ret;
-        }
-
-        // match several other cases
-        private List<Token> ImplicitDate(string text)
-        {
-            var ret = new List<Token>();
-            foreach (var regex in ImplicitDateList)
-            {
-                var matches = regex.Matches(text);
-                foreach (Match match in matches)
-                {
-                    ret.Add(new Token(match.Index, match.Index + match.Length));
-                }
-            }
-            return ret;
-        }
-
-        // process case like "三天前" "两个月前"
-        private List<Token> DurationWithBeforeAndAfter(string text)
-        {
-            var ret = new List<Token>();
-            var duration_er = _durationExtractor.Extract(text);
-            foreach (var er in duration_er)
-            {
-                var pos = (int)er.Start + (int)er.Length;
-                if (pos < text.Length)
-                {
-                    var tmp = text.Substring(pos, 1);
-                    if (tmp.Equals("前") || tmp.Equals("后"))
-                    {
-                        ret.Add(new Token(er.Start ?? 0, (er.Start + er.Length ?? 0) + 1));
-                    }
-                }
-            }
-            return ret;
-        }
-    }
+﻿using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Microsoft.Recognizers.Text.Number;
+using Microsoft.Recognizers.Text.Number.Chinese;
+
+namespace Microsoft.Recognizers.Text.DateTime.Chinese
+{
+    public class DateExtractorChs : IExtractor

```

Comment:
This is probably a separate issue, but in general classes in a namespace should not reflect the namespace itself.  In this case adding CHS to the end of the classes.